libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1324,6 +1324,12 @@ Json StandardCompiler::compileSolidity(StandardCompiler::InputsAndSettings _inpu
 
 	Json errors = std::move(_inputsAndSettings.errors);
 
+	if (_inputsAndSettings.debugInfoSelection.has_value() &&
+		_inputsAndSettings.debugInfoSelection->ethdebug &&
+		!isIRRequested(_inputsAndSettings.outputSelection)
+	)
+		errors.emplace_back(formatError(Error::Type::FatalError, "general", "debug.debugInfo setting for ethdebug only valid, if 'ir', 'irAst', 'irOptimized' or 'irOptimizedAst' where requested."));
```

Comment:
```suggestion
		errors.emplace_back(formatError(Error::Type::FatalError, "general", "debug.debugInfo setting for ethdebug only valid, if 'ir', 'irAst', 'irOptimized' or 'irOptimizedAst' were requested."));
```

But I think this would be a better wording:
```suggestion
		errors.emplace_back(formatError(Error::Type::FatalError, "general", "'ethdebug' can only be selected in 'settings.debug.debugInfo' when at least one of the IR outputs is selected."));
```

Reply:
Also, this is part of validation, so `parseInput()` seems like a better place for it. If you put it here, it will be completely ignored in Yul mode, without any indication for the user that it was ignored. You're also validating it *after* you have already passed it to `CompilerStack`.

Reply:
Finally, do we even need this validation? We don't do this for AST ID either, even though AFAIK it's Yul-only as well.

Reply:
I would like to have this validation. Or how could a user find out why something was not working?

Reply:
I'm all for strict validation in general, just this seems inconsistent with how we handle other debug info.

Also, I'm not sure if the condition is really that simple - I'd expect ethdebug info to be present in more artifacts than this, so it seems a bit artificial and leaky to me.